sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala

```diff
@@ -2720,4 +2704,26 @@ class HiveDDLSuite
       checkAnswer(sql("SHOW PARTITIONS ta_part"), Row("ts=10") :: Nil)
     }
   }
+
+  test("SPARK-31828: Retain table properties at CreateTableLikeCommand") {
+    val catalog = spark.sessionState.catalog
+    withTable("t1", "t2", "t3") {
+      sql(s"CREATE TABLE t1(c1 int) TBLPROPERTIES('k1'='v1', 'k2'='v2', 'totalNumberFiles'='meta')")
```

Comment:
Just in case, could you test all the prpos in `METASTORE_GENERATED_PROPERTIES`? Probably, its beetter to split it into two test units like add a new test unit `test("SPARK-31828: Filters out Hive metastore properties in CreateTableLikeCommand") {`.